File: collect_log_scripts/collect_syslogs.rb
#!/usr/bin/ruby

#
#	this script colelcts sytem logs to "/Users/Shared/Collected System Logs"
#
#	run this script with sudo
#


#REQUIRED RUBY MODULES
require 'fileutils'


#VARIBLES
collectionfolder='/Users/Shared/Collected System Logs'
system_logs='/var/log'


#MAKE DIRECTORY & FILE STRUCTURE SYSTEM LOGS
Dir.mkdir("#{collectionfolder}", 0755)


#COPY SYSTEM LOGS
FileUtils.copy_entry("/var/log", "#{collectionfolder}", :preserve => true)

File: collect_log_scripts/collect_syslogs.py
#!/usr/bin/env python 


#
#	this script colelcts sytem logs to "/Users/Shared/Collected System Logs"
#
#	run this script with sudo
#



#REQUIRED
import os, sys, shutil, distutils.dir_util

#VARIABLES
collction_folder="/Users/Shared/Collected System Logs"
system_logs="/var/log/"


#MAKE DIRECTOIRES & COPY SYSTEM LOG FILES
os.mkdir( collction_folder, 0777 );
#shutil.copytree(system_logs, collction_folder, symlinks=False, ignore=None)
distutils.dir_util.copy_tree(system_logs, collction_folder)

